content/software-security/what-are-containers/index.md

```diff
@@ -0,0 +1,64 @@
+---
+title: "What are Containers?"
+linktitle: "What are Containers?"
+description: "An overview of the structure, contents, and applications of container technology"
+lead: "An overview of the structure, contents, and applications of container technology"
+type: "article"
+date: 2023-10-17T20:02:23+00:00
+lastmod: 2023-10-24T18:38:29+00:00
+contributors: ["Michelle McAveety"]
+draft: false
+tags: ["Conceptual", "Overview"]
+images: []
+menu:
+  docs:
+    parent: "software-security"
+weight: 10
+toc: true
+---
+
+For decades, developers have sought out ways to maximize the utilization of their hardware. Dating back to the 1960s, virtual machines (VMs) were an early answer to this challenge, allowing a single computer to host multiple, isolated operating systems. VMs enable different guest users or processes to share physical infrastructure while keeping their concurrent operations separated. However, as VMs are both slow to initialize and resource-intensive, a modern solution arrived in the early 2000s: containers.
+
+*Containers* share a common *kernel* with each other, whereas multiple VMs each require their own virtual kernel. The kernel resides at the core of an operating system and facilitates activities between hardware and software. By sharing a kernel, containers run concurrently using the same infrastructure, providing the isolation benefits of VMs without added resource taxation. Containers have become increasingly popular for their relative ease of use, reproducibility, and portability in deploying applications across systems for a low resource cost.
+
+In this article, you will explore the structure of *container images*, the foundational unit behind containers, such as their key contents and the processes used to construct them. In addition, you will learn how containers operate on top of their container engine and associated infrastructure. You will also learn how to get started with using containers, including building, choosing and deploying container images for your applications.
+
+
+## Structure of a Container Image
+
+In order to build a container for your application, you will need to start with a *container image*. A container image is a static, immutable filesystem bundle that serves as a blueprint you can use to build containers. Inside every container image is a curated collection of the files, dependencies and code needed to run an application. At runtime, when a container is built from an image, the resultant container inherits all characteristics of the container image it is instantiated from.
+<!-- Place a graphic here showing the result of instantiating a container -->
```

Comment:
will work with Travers here and update article graphics after I finalize discussions there